xknx/management/management.py

```diff
@@ -116,6 +128,48 @@ async def connection(
         finally:
             await self.disconnect(address)
 
+    async def send_broadcast(self, telegram: Telegram) -> None:
+        """Send a broadcast message."""
+
+        if isinstance(telegram.tpci, TDataBroadcast):
+            await self.xknx.cemi_handler.send_telegram(telegram)
+        else:
+            raise TypeError(
+                f"Can only send broadcast telegrams, not {type(telegram.tpci)}."
+            )
+
+    @asynccontextmanager
+    async def broadcast(self) -> AsyncIterator[BroadcastContext]:
+        """Provide a broadcast context."""
+        context = BroadcastContext()
+        self._broadcast_contexts.add(context)
+        try:
+            yield context
+        finally:
+            self._broadcast_contexts.remove(context)
+
+
+class BroadcastContext:
+    """Class providing broadcast contexts."""
+
+    def __init__(self) -> None:
+        """Initialize BroadcastContext class."""
+        self.queue: asyncio.Queue[Telegram] = asyncio.Queue()
+
+    async def receive(
+        self, timeout: float | None = 3
+    ) -> AsyncGenerator[Telegram, None]:
+        """Receive telegrams from the broadcast context."""
+        try:
+            async with asyncio_timeout(timeout):
+                while True:
+                    try:
+                        yield await self.queue.get()
+                    except GeneratorExit:
+                        return
```

Comment:
I think we could test this by injecting 2 received telegrams (multiple devices in programming mode) at `nm_individual_address_read(raise_if_multiple=True)`.

Reply:
I added a test for to check for the exception, but I don't think that will also cover the `GeneratorExit` exception.

Reply:
Sure, looking at Codecov it is already covered.